stix2generator/stix21_registry.json

```diff
@@ -2771,15 +2778,15 @@
     },
     "indicator-type-ov": {
         "type": "string",
-        "oneOf": [
-            "anomalous-activity",
-            "anonymization",
-            "benign",
-            "compromised",
-            "malicious-activity",
-            "attribution",
-            "unknown"
-        ]
+        "oneOf": {
+            "10": "anomalous-activity",
+            "10": "anonymization",
+            "10": "benign",
+            "10": "compromised",
+            "10": "malicious-activity",
+            "10": "attribution",
+            "40": "unknown"
+        }
```

Comment:
This JSON object has duplicate keys, so it wouldn't work.  Weights need not be distinct, therefore we can't use them as keys.  For full generality, the things being weighted would need to be arbitrary specs (not just strings), so they can't be keys either.  If the value of "oneOf" were a list of two-element [weight, spec] lists, it would be ambiguous since all non-object JSON values already have the literal interpretation as a spec which generates that fixed value.  E.g. does a [weight,spec] value mean to generate that fixed list, or something else?  I think an array value for "oneOf" will need to retain its current semantics.  The only way I can think to have an unambiguous alternative structure is to add support for an object value here.  E.g. {"weights": [...], "specs": [...]}.

But there are currently no probabilities anywhere in the spec structure, and I kind of like the idea of keeping it more declarative and not adding behavioral complexity like specifying probabilities.  Different users might want to use different probabilities, so it doesn't seem like the kind of thing which should be baked into the specifications.  Or it should at least be overridable somehow.  One could also imagine defining probabilities and/or distributions over more than just "oneOf" specs.  One could define probabilities for optional properties of objects, and distributions for boolean specs (e.g. probability of true vs false), maybe numeric ranges, ...  A distribution doesn't make sense for optional properties, since there is no need to pick any of them.  In fact, the "minimize_ref_properties" config setting could be re-conceptualized as a shortcut for setting p=0 for all ref properties.  A distribution is necessary for something like a boolean spec, since either a true or false value must *always* be generated in that case.  Turning down the probability for one outcome must necessarily push the probability up for other outcomes.

Anyway, providing probabilities separately from the registry, e.g. as a config item, would require the ability to specify what to apply the probabilities to.  E.g. to either refer to a component of a specification, or a component of a JSON value being generated.  Maybe a JSONPath-like idea?  Hmmmm.

Reply:
> This JSON object has duplicate keys, so it wouldn't work. 

Of course, I was trying to do it the "other way" - i.e., the probability being the value, but that didn't work when I tried it for indicator/valid_until - as you also say.  I might try the "unambiguous alternative structure" you suggest.  If it works, we can move on to your hesitancy about putting probabilities in the spec structure.

I was thinking that if different users wanted different probabilities, we could use the override capability that already exists.  We could add an argument to the stix-generator command line to take in the override file.  

I understand this is not a complete solution for probabilistic choices - but maybe a stop-gap solution?

I'm open to a config item - if you want to explore it.